go/arrow/array/float16.go

```diff
@@ -87,10 +87,16 @@ func (a *Float16) GetOneForMarshal(i int) interface{} {
 func (a *Float16) MarshalJSON() ([]byte, error) {
 	vals := make([]interface{}, a.Len())
 	for i, v := range a.values {
-		if a.IsValid(i) {
-			vals[i] = v.Float32()
-		} else {
+		if !a.IsValid(i) {
 			vals[i] = nil
+			continue
+		}
+
+		switch {
+		case v.IsNaN():
+			vals[i] = "NaN"
+		default:
+			vals[i] = v.Float32()
 		}
```

Comment:
no check for `Inf`?

Reply:
Yeah. I had a doubt in this. If I use float16.Inf(), I assume that is "+Inf". What for "-Inf" then?

Reply:
We should add an `IsInf` method to `float16.Num` which can do `return n.bits & 0x7c00`. After that you can simply check the sign bit via `Signbit()` to determine if it's `-Inf` or `+Inf`.